packages/contracts-bedrock/src/periphery/faucet/Faucet.sol

```diff
@@ -88,7 +94,8 @@ contract Faucet {
     /// @notice Drips ETH to a recipient account.
     /// @param _params Drip parameters.
     /// @param _auth   Authentication parameters.
-    function drip(DripParameters memory _params, AuthParameters memory _auth) public {
+    /// @param _bridge Bridge address.
+    function drip(DripParameters memory _params, AuthParameters memory _auth, address _bridge) public {
```

Comment:
Alternatively instead of adding custom code here for bridging, could also just modify `DripParameters` to include a `data` field. Would keep the faucet generic.

Reply:
would the data field still be of type `address`

would this change the logic of the function body?

Reply:
Data field would be type `bytes`, so:

```
struct DripParameters {
        address payable recipient;
        bytes data;
        bytes32 nonce;
}
```

Then honestly instead of `SafeSend` you could just:

```
_params.recipient.call{value: config.amount}(_params.data);
```

Reply:
ah i see, moving bridge info from a param for the drip function to DripParameters could work

I think I'm still not sure about the execution logic. When initializing the l1bridge I want an an address type value to pass in (`IL1StandardBridge(_bridge)`). And then I want to be able to call depositETHTo with that

Could modify DripParameters to 

```
struct DripParameters {
      address payable recipient;
      address bridge;
      bytes32 nonce;
}
```
And for the logic in the function:
```
if (_params.bridge == address(0)) {
      // Use SafeSend if no bridge address is provided
      new SafeSend{ value: config.amount }(_params.recipient);
} else {
      // Execute a bridging of ETH to the recipient account.
      IL1StandardBridge l1Bridge = IL1StandardBridge(_params.bridge);
      l1Bridge.depositETHTo{value: config.amount}(_params.recipient, 200000, "");
  }
```

Reply:
Ah I see what you mean - so you would pass the function header and necessary params (i.e. `depositETHTo` or no value inside the `data` variable). Couple questions on doing that

- If we used `_params.recipient.call{value: config.amount}(_params.data);` would no value for `_params.data` or a zeroAddress value result in a transfer call? I'm trying to understand how that code behaves in the case in where we don't want to bridge
- I'm also unfamiliar with how to properly encode a function call with params into the `bytes data` to have this work. Going to give it a try tomorrow